pydoc/developer/TIPS.md

```diff
@@ -0,0 +1,39 @@
+.. _build_tips:
+
+Speeding up Arkouda Compilation
+===============================
+
+## Environment Variables to _Always_ Set
+
+This section has a couple of environment variables that should _always_ be set/unset if you are looking for compilation time improvements.
+
+1. `export ARKOUDA_QUICK_COMPILE=1`
+    - this env variable turns off a couple of optimizations, speeding up builds
+    - performance will be worse when compiled with this env variable, but that typically does not matter when doing development/correctness tests
+7. `unset ARKOUDA_DEVELOPER`
+    - this env variable should be unset to get faster builds because it currently overrides `ARKOUDA_QUICK_COMPILE`, so if you have both set, you will not get the quick compile speed up 
+    - this env variable enables checks in the Chapel generated code, so performance takes a bit of a hit
+8. `unset CHPL_DEVELOPER`
+     - if this was set at the time that the chpl compiler was set, then the chpl compiler should be rebuilt with this unset
+    - if you are using a homebrew installation of Chapel, you shouldn't need to worry about rebuilding the chpl compiler
+
+## Using the Modular Build System
+
+This is the single biggest way to save on compilation time, but is a bit more involved than the others. For reference, a full build on my machine takes ~500 seconds, but with minimal modules, it only takes ~30 seconds.
```

Comment:
I think what makes the most sense here is to keep the documentation for the modular build stuff in the developer section, removing that detail from the regular docs, but keep the part about adding new modules into the build process, as those seem like two different things and I see the compilation improvement as more developer-oriented. Sound good?

Reply:
Hmm, actually, on second thought, these seem like pretty different docs to me, I don't think there is a ton of overlap since the developer docs are pretty much entirely dedicated to speeding up compilation, whereas the others are just around the modular build usage. I think they seem sufficiently different enough that I think we should keep both and I don't think we should combine them either because they are definitely for different audiences... Maybe I'll just add links to each to the other? Like in the dev docs say like "for more general information, see the other docs" and then in the existing docs say like "for tips on speeding up compilation, see the dev docs".

How does that sound?

Reply:
that sounds good!